build.zig

```diff
@@ -61,73 +58,128 @@ pub fn build(b: *Build) !void {
         .{
             .name = "UI",
             .root_source_file = "samples/ui/src/main.zig",
-            .resource_dir = "samples/ui/data",
+            .resource_dir = b.path("samples/ui/data"),
         },
     }) |sample| {
-        // Module structure:
-        //  root = src/orca.zig
-        //  app = sample/src/main.zig
-        // @Incomplete: modify orca.zig so it can be used as a module instead of taking over the root
-
         const root = b.createModule(.{
-            .root_source_file = b.path("src/orca.zig"),
+            .root_source_file = b.path(sample.root_source_file),
             .target = wasm_target,
             .optimize = optimize,
         });
-        const user_root = b.createModule(.{
-            .root_source_file = b.path(sample.root_source_file),
-        });
-        root.addImport("user_root", user_root);
-        user_root.addImport("orca", root);
+        root.addImport("orca", orca_mod);
 
-        const app_wasm = b.addExecutable(.{
+        const app_wasm = addApplication(b, .{
             .name = sample.name,
             .root_module = root,
+            .sdk_path = sdk_path,
         });
-        app_wasm.entry = .disabled; // See https://github.com/ziglang/zig/pull/17815
-        app_wasm.rdynamic = true;
-        app_wasm.addObjectFile(sdk_path.path(b, "bin/liborca_wasm.a"));
-        app_wasm.addObjectFile(sdk_path.path(b, "orca-libc/lib/libc.o"));
-        app_wasm.addObjectFile(sdk_path.path(b, "orca-libc/lib/libc.a"));
-        app_wasm.addObjectFile(sdk_path.path(b, "orca-libc/lib/crt1.o"));
-
         sample_check_step.dependOn(&app_wasm.step);
 
-        const run_bundle = b.addSystemCommand(&.{
-            "orca",      "bundle",
-            "--name",    sample.name,
-            "--version", sdk_version,
-        });
-        if (sample.icon) |icon| {
-            run_bundle.addArg("--icon");
-            run_bundle.addDirectoryArg(b.path(icon));
-        }
-        if (sample.resource_dir) |res_dir| {
-            run_bundle.addArg("--resource-dir");
-            run_bundle.addDirectoryArg(b.path(res_dir));
-        }
-        run_bundle.addArg("--out-dir");
-        const bundle_output = run_bundle.addOutputDirectoryArg(sample.name);
-        run_bundle.addArtifactArg(app_wasm);
-
         sample_step.dependOn(
-            &b.addInstallDirectory(.{
-                .source_dir = bundle_output.path(b, sample.name), // orca bundle creates a subdir with the app name
-                .install_dir = .prefix,
-                .install_subdir = sample.name,
+            &addInstallApplication(b, .{
+                .app = app_wasm,
+                .icon = sample.icon,
+                .resource_dir = sample.resource_dir,
             }).step,
         );
     }
 }
+
+pub fn target(b: *Build) Build.ResolvedTarget {
+    return b.resolveTargetQuery(.{
+        .cpu_arch = .wasm32,
+        .os_tag = .freestanding,
+        .cpu_features_add = std.Target.wasm.featureSet(&.{.bulk_memory}),
```

Comment:
fwiwi orca libc and sdk are also built with `std.Target.wasm.Feature.nontrapping_fptoint`, but I think this is fine too